registry-scanner/pkg/image/image.go

```diff
@@ -152,6 +168,49 @@ func (img *ContainerImage) DiffersFrom(other *ContainerImage, checkVersion bool)
 	return img.RegistryURL != other.RegistryURL || img.ImageName != other.ImageName || (checkVersion && img.ImageTag.TagName != other.ImageTag.TagName)
 }
 
+// Clone creates a deep copy of the ContainerImage object.
+func (img *ContainerImage) Clone() *ContainerImage {
+	if img == nil {
+		return nil
+	}
+	clone := &ContainerImage{
+		RegistryURL:           img.RegistryURL,
+		ImageName:             img.ImageName,
+		ImageAlias:            img.ImageAlias,
+		UpdateStrategy:        img.UpdateStrategy,
+		ForceUpdate:           img.ForceUpdate,
+		AllowTags:             img.AllowTags,
+		PullSecret:            img.PullSecret,
+		HelmParamImageName:    img.HelmParamImageName,
+		HelmParamImageVersion: img.HelmParamImageVersion,
+		original:              img.original,
+	}
+
+	if img.ImageTag != nil {
+		clone.ImageTag = &tag.ImageTag{
+			TagName:   img.ImageTag.TagName,
+			TagDigest: img.ImageTag.TagDigest,
```

Comment:
how about the field `TagDate   *time.Time` in `ImageTag`?

Reply:
Great! Thank you